packageurl.go

```diff
@@ -371,22 +367,17 @@ func validQualifierKey(key string) bool {
 	return QualifierKeyPattern.MatchString(key)
 }
 
-// Make any purl type-specific adjustments to the url encoding.
+// pathEscape Make any purl type-specific adjustments to the url encoding.
 // See https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#character-encoding
-func PurlPathEscape(s string) string {
-	return Encode(s, "@#? ")
-}
-
-func Encode(s string, charsToEncode string) string {
+func pathEscape(s string) string {
 	var t strings.Builder
 	for _, c := range s {
-		if strings.ContainsRune(charsToEncode, c) || c > unicode.MaxASCII {
-			for _, b := range []byte(string(c)) {
-				t.WriteByte('%')
-				t.WriteByte(upperhex[b>>4])
-				t.WriteByte(upperhex[b&15])
-			}
-		} else {
+		switch {
+		case c == '@':
+			t.WriteString("%40")
+		case c == '?' || c == '#' || c == ' ' || c > unicode.MaxASCII:
```

Comment:
If i understood correctly url.PathEscape is not exactly what we need for encoding.
I tried to look for the base encoding function used by the `url` library but it is baked in so i had to export and write my own encoding function..
From url/url.go
```
		case encodePathSegment: // §3.3
			// The RFC allows : @ & = + $ but saves / ; , for assigning
			// meaning to individual path segments.
			return c == '/' || c == ';' || c == ',' || c == '?'
```
So i think `#` must be encoded explicit manner

url/url.go encoding part ..
```
		case shouldEscape(c, mode):
			t[j] = '%'
			t[j+1] = upperhex[c>>4]
			t[j+2] = upperhex[c&15]
			j += 3
			```
PS: 
url/url.go query encode encoding a wider range '$', '&', '+', ',', '/', ':', ';', '=', '?', '@':

Reply:
I have a test that covers these characters: https://github.com/anchore/packageurl-go/pull/4/files#diff-dd9ce18846ff2135af320932942f9858df9edb7fd625922aea9a526e2ab4b0ceR327-R328